ERCS/erc-7846.md

```diff
@@ -12,129 +12,194 @@ created: 2024-12-15
 
 ## Specification
 
-### `wallet_connect`
+### JSON-RPC Methods
+
+#### `wallet_connect`
 
-Request the user to connect a single account and optionally confirm chain support and add capabilities.
+Requests to connect account(s) with optional capabilities.
 
-#### RPC Specification
+##### Request
 
-For each chain requested by the app, wallets MUST return a mapped capabilities object if the chain is supported and SHOULD return an empty object if no capabilities exist. Wallets MUST NOT return mapped capabilities objects for chains they do not support. If an app does not declare chains they would like to confirm support for, the wallet can return any chains it wishes to declare support for.
+```ts
+type Request = {
+  method: 'wallet_connect',
+  params: [{
+    // JSON-RPC method version.
+    version: string;
+    // Optional capabilities to request (e.g. Sign In With Ethereum).
+    capabilities?: Record<string, unknown>;
+  }]
+}
+```
 
-```typescript
-type WalletConnectParams = [{
-  version: string;
-  capabilities?: Record<string,any>; // optional connection capabilities
-}]
+##### Response
 
-type WalletConnectResult = {
+List of connected accounts with their associated capabilities.
+
+```ts
+type Response = {
   accounts: {
-    address: `0x${string}`; // connected account address
-    capabilities: Record<string,any>; // results of this connection request's connection capabilities
+    // Address of the connected account.
+    address: `0x${string}`;
+    // Capabilities granted that is associated with this account.
+    capabilities: Record<string, unknown>;
   }[]
 }
 ```
 
-#### Example Parameters
-
-```json
-[{
-  "version": "1",
-  "capabilities": {
-    "signInWithEthereum": {
-      "nonce": "12345678",
-      "chainId": "0x1"
+##### Example
+
+```ts
+const response = await provider.request({
+  method: 'wallet_connect',
+  params: [{
+    version: '1',
+    capabilities: {
+      signInWithEthereum: {
+        nonce: '12345678',
+        chainId: '0x1'
+      }
     }
-  }
-}]
+  }]
+})
+/**
+ * {
+ *   accounts: [
+ *     {
+ *       address: '0x...',
+ *       capabilities: {
+ *         signInWithEthereum: {
+ *           message: 'app.com wants you to sign in with your Ethereum account:\n0x...',
+ *           signature: '0x...'
+ *         }
+ *       }
+ *     }
+ *   ]
+ * }
+ */
 ```
 
-#### Example Result
-
-```json
-{
-  "accounts": [
-    {
-      "address": "0x...",
-      "capabilities": {
-        "signInWithEthereum": {
-          "message": "app.com wants you to sign in with your Ethereum account:\n0x...",
-          "signature": "0x..."
-        }
-      }
-    }
-  ]
+#### `wallet_disconnect`
+
+Disconnects connected account(s).
+
+- The wallet SHOULD revoke any capabilities associated with the account(s) that were granted upon connection via `wallet_connect`.
```

Comment:
Need to think more on this. May be comingling things too much. If I log out of Coinbase, that doesn't mean I want to disconnect my connection to my bank account if I set that up on login (attempt at analogy to combining with a session/permission example).